pkg/apis/storage/types.go

```diff
@@ -310,6 +327,36 @@ type CSIDriverSpec struct {
 	VolumeLifecycleModes []VolumeLifecycleMode
 }
 
+// FSGroupPolicy specifies if a CSI Driver supports modifying
+// volume ownership and permissions of the volume to be mounted.
+// More modes may be added in the future.
+type FSGroupPolicy string
+
+const (
+	// Heuristic indicates that each volume should be examined
+	// to determine if the volume ownership and permissions
+	// should be modified.
```

Comment:
Can we be more explicit about the conditions where this will be true, similar to the comment at https://github.com/kubernetes/kubernetes/pull/92001/files#diff-7dc0dea2b32ece49304df77954ffb69aR279. In fact, it should probably be copied verbatim or only specified in one place to avoid them getting out of date.

Reply:
I've moved the descriptions into the enum types, and mentioned that these should be examined for the behaviour of the fsGroupPolicy field.